src/solvers/minilp.rs

```diff
@@ -70,6 +78,27 @@ impl SolverModel for MiniLpProblem {
             }),
         }
     }
+
+    fn add_constraint(&mut self, c: Constraint) -> ConstraintReference {
+        let coefficients: Vec<(minilp::Variable, f64)> = c
+            .expression
+            .linear
+            .coefficients
+            .iter()
+            .map(|(var, &coeff)| (self.variables[var.index()], coeff))
+            .collect();
+        let op = match c.is_equality {
+            true => minilp::ComparisonOp::Eq,
+            false => minilp::ComparisonOp::Le,
+        };
+        let constant = -c.expression.constant;
+        self.problem.add_constraint(&coefficients, op, constant);
+        self.n_constraints += 1;
+
+        ConstraintReference {
+            index: self.n_constraints - 1,
+        }
+    }
```

Comment:
We probably shouldn't duplicate code between `add_constraint` and `with`. We can have a default method implementation in `SolverModel` for with that internally uses add_constraint.